src/goto-instrument/contracts/dynamic-frames/dfcc_infer_loop_assigns.cpp

```diff
@@ -196,7 +270,9 @@ assignst dfcc_infer_loop_assigns(
     {
       address_of_exprt address_of_expr(expr);
       address_of_expr.add_source_location() = expr.source_location();
-      if(!is_constant(address_of_expr))
+      // Widen assigns targets to object_whole if `expr` is a dereference or
+      // with constant address.
+      if(expr.id() == ID_dereference || !is_constant(address_of_expr))
```

Comment:
I would appreciate if this change could be moved to a commit of its own with an accompanying test that demonstrates the difference for this change may impact existing users.